### DISCUSSION TITLE
How to setValue of FieldArray?

### DISCUSSION THREAD

**Discussion by difaananda40** [react-hook-form/react-hook-form#2429] [Q&A] [answered]
Hallo,
I have a fieldarray, and i want to update the value on useEffect.
But when i try setValue('name', [...]) , it doesn't change anything.

**Comment by GabrielGomesBarbosa** [comment]
did you try `reset()` method ?

**Comment by bluebill1049** [comment, marked as answer]
setValue accept `fieldName`, or array of `fieldNames`. we don't accept `key`.

**Reply by difaananda40** [reply]
I tried this, and nothing happen
![image](https://user-images.githubusercontent.com/32413048/88988161-a7688500-d30a-11ea-8c0e-0ce26adac04f.png)

**Reply by bluebill1049** [reply]
- `fieldName` => `test[0].test`
- `keyName` => `test`

you can try with `reset()` which support an entire object.

**Reply by difaananda40** [reply]
okay, i'll try that

**Reply by difaananda40** [reply]
yes, reset is good, but can i import reset from useFormContext? or just provide it to props?

**Reply by bluebill1049** [reply]
you can use reset from provider

**Reply by difaananda40** [reply]
ah okay,
when using reset, I just want set the selected field, and I don't want to reset other fields, is it possible?

**Reply by bluebill1049** [reply]
reset(getValues())

**Reply by difaananda40** [reply]
it's a good library, thanks for helping me :+1:

**Reply by perdanafm** [reply]
Hii, can u gave me an example of this ? i was stuck on this and have no answer

**Reply by Moshyfawn** [reply]
> Hii, can u gave me an example of this ? i was stuck on this and have no answer

Check out [the `resetField` method](https://react-hook-form.com/api/useform/resetfield/#main). It might be more suitable for your needs

**Reply by byteab** [reply]
I'm using `reset({fieldName: [{}, {}]})`, but still it only set the first object of the arrary
